packages/engine/src/worker/mod.rs

```diff
@@ -327,62 +396,52 @@ impl WorkerController {
         Ok(())
     }
 
-    async fn finish_task_from_runner_msg(
-        &mut self,
-        sim_id: SimulationShortId,
-        msg: RunnerTaskMsg,
-        source: Language,
-    ) -> Result<()> {
-        self.finish_task(msg.task_id, sim_id, source, msg.payload, msg.shared_store)
-            .await
-    }
-
-    fn inbound_from_task_msg(
-        task_id: TaskId,
-        package_id: PackageId,
-        shared_store: TaskSharedStore,
-        task_msg: TaskMessage,
-    ) -> InboundToRunnerMsgPayload {
-        InboundToRunnerMsgPayload::TaskMsg(RunnerTaskMsg {
-            task_id,
-            package_id,
-            shared_store,
-            payload: task_msg,
-        })
-    }
-
-    /// TODO: DOC
+    /// Handles a runner message based on the target of the current task.
+    ///
+    ///   Depending on the [`target`], the following actions are executed:
```

Comment:
Calls the task's WorkerHandler impl to create the follow-up task, and depending on the target ....

Reply:
I think we need to rework this comment, the first line doesn't really describe what the function does.

It's more like
> Calls the task's [`WorkerHandler`] implementation to create the follow-up task and depending on the target of that, it handles the given message.
> 
> Depending on the [`target`], the following ...